Simplenote/src/androidTest/java/com/automattic/simplenote/uitests/login/LoginTests.kt

```diff
@@ -0,0 +1,95 @@
+package com.automattic.simplenote.uitests.login
+
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.MediumTest
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import com.automattic.simplenote.BaseUITest
+import com.automattic.simplenote.R
+import com.automattic.simplenote.utils.Constants
+import org.junit.Test
+import org.junit.runner.RunWith
+import androidx.test.espresso.matcher.ViewMatchers.withClassName
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import com.automattic.simplenote.utils.TestDriver
+import org.hamcrest.Matchers.*
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.isClickable
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import org.junit.Assert
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@MediumTest
+class LoginTests : BaseUITest() {
+
+    @Test
+    fun testLoginWithWrongPasswordFails() {
+        onView(withId(R.id.button_login))
+            .check(matches(isClickable()))
+            .perform(click())
+
+        onView(withId(R.id.button_email))
+            .check(matches(isClickable()))
+            .perform(click())
+
+        onView(allOf(isDescendantOfA(withId(R.id.input_email)),
+            withParent(withClassName(endsWith("android.widget.FrameLayout"))),
```

Comment:
Actually I was unable to identify the text field otherwise. Initially I tried using _withClassName(endsWith("android.widget.EditText"))_ but it did not work. Using the _withParent_ matcher however helped.

Reply:
Hm, that's interesting 🤔 In my case, removing them had no effect - objects were still located.